django_recaptcha/static/django_recaptcha/js/widget_v3.js

```diff
@@ -0,0 +1,50 @@
+(function () {
+
+  function main() {
+    grecaptcha.ready(function () {
+      for (const captchaElement of document.querySelectorAll(".g-recaptcha")) {
+
+        const widgetUuid = captchaElement.getAttribute("data-widget-uuid");
+        if (!widgetUuid) {
+          console.warn("reCAPTCHA widget with missing UUID");
+          break;
+        }
+        console.log(`found reCAPTCHA widget with UUID '${widgetUuid}'`);
+
+        const formElement = captchaElement.closest("form");
+        if (!formElement) {
+          console.warn("reCAPTCHA widget with UUID '" + widgetUuid + "' is not part of a form");
+          break;
+        }
+
+        const publicKey = captchaElement.getAttribute("data-sitekey");
+        if (publicKey === null) {
+          console.warn("public key missing missing from reCAPTCHA widget with UUID '" + widgetUuid + "'");
+          break;
+        }
+
+        const action = {};
+        Object.assign(action, {action: captchaElement.getAttribute("data-action")});
+
+        formElement.addEventListener("submit", function (event) {
+          event.preventDefault();
+          grecaptcha.execute(publicKey, action)
```

Comment:
**Suggestion:** make it more clear what this variable is used for and avoid potential confusion

```suggestion
        const config = {};
        Object.assign(config, {action: captchaElement.getAttribute("data-action")});

        formElement.addEventListener("submit", function (event) {
          event.preventDefault();
          grecaptcha.execute(publicKey, config)
```

Reply:
I actually rewrote this part several times before, but eventually decided on using the variable name matching [1]. Using `Object.assign()` also isn't the most clear, so I changed that as well in my new commit, with an additional comment to explain the situation.